epc_data_analysis/pipeline/easy_plotting.py

```diff
@@ -0,0 +1,606 @@
+        If None, plot is not saved.
+        If "" or "auto", filename is generated automatically.
+
+    plot_kind : {"hist", "bar"}, default="hist"
+        Type of plot.
+
+    plotting_colors : list, str, None, default=None
+        Ordered list of colors or color map to use when plotting feature 2.
+        If list, use list of colors.
+        If str, use corresponding matplotlib color map.
+        If None, use default color list.
+
+    y_ticklabel_type : {'', 'm', 'k' or '%'}, default=None
+        Label for yticklabel, e.g. 'k' when displaying numbers
+        in more compact way for easier readability (50000 --> 50k)."""
+
+    # Remove all samples for which feature 1 or feature 2 is NaN.
+    df = df[df[feature_1].notna()]
+    df = df[df[feature_2].notna()]
+
+    # Get set of values/subcategories for features.
+    feature_1_values = list(set(df[feature_1].sort_index()))
+    feature_2_values = list(set(df[feature_2].sort_index()))
+
+    # Set order for feature 1 values/subcategories
+    if feature_1_order is not None:
+        feature_1_values = feature_1_order
+
+    # Create a feature-bar dict
+    feat_bar_dict = {}
+
+    # For every feature 2 value/subcategory, get feature 1 values
+    # e.g. for every tenure type, get windows energy efficiencies
+    for feat2 in feature_2_values:
+        dataset_of_interest = df.loc[df[feature_2] == feat2][feature_1]
+        data_of_interest = dataset_of_interest.value_counts()
+        feat_bar_dict[feat2] = data_of_interest
+
+    # Save feature 2 subcategories by feature 1 subcategories as dataframe
+    subcat_by_subcat = pd.DataFrame(feat_bar_dict, index=feature_1_values)
+
+    # If feature 2 order is given, rearrange
+    if feature_2_order is not None:
+        subcat_by_subcat = subcat_by_subcat[feature_2_order]
+
+    # If not defined, set default colors for plotting
+    if plotting_colors is None:
+        plotting_colors = ["green", "greenyellow", "yellow", "orange", "red"]
+
+    # Use given colormap
+    if isinstance(plotting_colors, str):
+        cmap = plotting_colors
+        subcat_by_subcat.plot(kind=plot_kind, cmap=cmap)  # recommended RdYlGn
+
+    # or: use given color list
+    elif isinstance(plotting_colors, list):
+        subcat_by_subcat.plot(kind=plot_kind, color=plotting_colors)
+
+    else:
+        raise IOError("Invalid plotting_colors '{}'.".format(plotting_colors))
+
+    # Get updated yticklabels
+    ax = plt.gca()
+    yticklabels, ax, _, _ = get_readable_tick_labels(plt, y_ticklabel_type, "y")
+    ax.set_yticklabels(yticklabels)
+
+    # Set plot title
+    if plot_title is None:
+        plot_title = feature_2 + " by " + feature_1
+
+    # Describe plot with title and axes
+    plt.title(plot_title)
+    plt.ylabel(y_label)
+    plt.xlabel(x_label)
+    plt.xticks(rotation=0)
+
+    # Save figure
+    save_figure(plt, save_filename, plot_title)
+
+    # Show plot
+    plt.show()
+
+
+def get_pearson_correlation(df, feature_1, feature_2):
```

Comment:
Could use `scipy.stats.pearsonr`?

Reply:
I changed it to `scipy.stats.pearsonr` but I kept the function for now.